okapi-core/src/test/java/org/folio/okapi/ProxyTest.java

```diff
@@ -4344,4 +4354,170 @@ public void testManyModules(TestContext context) throws IOException {
+        + "  \"filters\" : [ {" + LS
+        + "    \"methods\" : [ \"GET\", \"POST\" ]," + LS
+        + "    \"path\" : \"/\"," + LS
+        + "    \"phase\" : \"post\"," + LS
+        + "    \"type\" : \"request-response\"," + LS
+        + "    \"permissionsRequired\" : [ ]" + LS
+        + "  } ]" + LS
+        + "}";
+    c = api.createRestAssured3();
+    c.given()
+        .header("Content-Type", "application/json")
+        .body(docRequestPost).post("/_/proxy/modules").then().statusCode(201)
+        .extract().response();
+    Assert.assertTrue("raml: " + c.getLastReport().toString(),
+        c.getLastReport().isEmpty());
+
+    nodeDoc1 = "{" + LS
+        + "  \"instId\" : \"localhost-2\"," + LS
+        + "  \"srvcId\" : \"module-post-1.0.0\"," + LS
+        + "  \"url\" : \"http://localhost:" + Integer.toString(portPost) + "\"" + LS
+        + "}";
+    given().header("Content-Type", "application/json")
+        .body(nodeDoc1).post("/_/discovery/modules")
+        .then().statusCode(201);
+
+    final String docTimer_1_0_0 = "{" + LS
+        + "  \"id\" : \"timer-module-1.0.0\"," + LS
+        + "  \"name\" : \"timer module\"," + LS
+        + "  \"provides\" : [ {" + LS
+        + "    \"id\" : \"echo\"," + LS
+        + "    \"version\" : \"1.0\"," + LS
+        + "    \"handlers\" : [ {" + LS
+        + "      \"methods\" : [ \"POST\" ]," + LS
+        + "      \"pathPattern\" : \"/echo\"," + LS
+        + "      \"permissionsRequired\" : [ ]" + LS
+        + "    } ]" + LS
+        + "  } ]," + LS
+        + "  \"requires\" : [ ]" + LS
+        + "}";
+    given()
+        .header("Content-Type", "application/json")
+        .body(docTimer_1_0_0).post("/_/proxy/modules").then().statusCode(201)
+        .extract().response();
+
+    nodeDoc1 = "{" + LS
+        + "  \"instId\" : \"localhost-3\"," + LS
+        + "  \"srvcId\" : \"timer-module-1.0.0\"," + LS
+        + "  \"url\" : \"http://localhost:" + Integer.toString(portTimer) + "\"" + LS
+        + "}";
+    given().header("Content-Type", "application/json")
+        .body(nodeDoc1).post("/_/discovery/modules")
+        .then().statusCode(201);
+
+    JsonArray installReq = new JsonArray().add(new JsonObject().put("id",  "timer-module-1.0.0").put("action", "enable"));
+    c = api.createRestAssured3();
+    c.given()
+        .header("Content-Type", "application/json")
+        .body(installReq.encode())
+        .post("/_/proxy/tenants/" + okapiTenant + "/install?deploy=true")
+        .then().statusCode(200).log().ifValidationFails()
+        .body(equalTo(installReq.encodePrettily()));
+    Assert.assertTrue(
+        "raml: " + c.getLastReport().toString(),
+        c.getLastReport().isEmpty());
+
+    given().header("X-Okapi-Tenant", okapiTenant)
+        .header("Content-Type", "text/plain")
+        .body("Okapi").post("/echo")
+        .then().statusCode(200)
+        .header("Content-Type", "text/plain; charset=ISO-8859-1")
+        .body(equalTo("Okapi"));
```

Comment:
Consider to add a line to check non-existing gzip header `.header("Content-Encoding", nullValue())`

Reply:
ok